### ISSUE TITLE
[AntennaPod] Quits download via Wifi when using Blokada

### ISSUE THREAD

**Issue by Krawei** [blokadaorg/blokada#216]
Dear team,

when I'm activating Blokada and want to download podcasts with AntennaPod I can't do it via Wifi anymore.

AntennaPod asks me to download my  podcast audio files via cellular/4G instead of using my working Wifi connection.
There are no "blocked xyz" event messages from blokdada while this, but it works again when I'm disabling Blokada.

I hope I made my issue clear. Please fix this pity behaviour :).

**Comment by noelkra** [comment]
Hello Krawei
We're sorry to hear that! Could you please tell us which OS you use and what filter-lists you are using so we can reproduce it :) 
Regards. 
Noël
P. S.: Please use the issue layout for issues to make it easier for us to get an overview of the problem.

**Comment by Krawei** [comment]
Hello Noël,

I'm using these filter lists:

AdZHosts (recommended)
Mother of all AdBlocking
"vereinheitlichte hosts-liste" (https://raw.githubunsercontent.com/StevenBlack/hosts/master/hosts)
HPHosts  ads hosts-Datei
MVPS hosts-Datei
Adaway hosts-Datei
Dan Pollock's hosts-Datei
Peter Lowe hosts-Datei
Social hosts file
Coin Blocker Lists

**Comment by Krawei** [comment]
Here is the requested issue layout:

Steps to reproduce

While connected to a Wifi and running Blockada in the background, open AntennaPod => update episodes => seek for an interesting episode => click on the download arrow.

Expected behaviour
Antennapod should download the episode without any issues.

Actual behaviour
AntennaPod asks for downloading the episode via cellular/4G. That happends when there isn't a current Wifi connection.
    Blokada version
    3.4

    Device model
    BQ Aquaris X

    Android version / ROM
    8.1.0

Questions

    How long have you been using Blokada?
    Since two weeks

    How long time ago has the problem started to happen?
    Since the beginning.

    How often does this problem happen to you?
    Everytime.

**Comment by noelkra** [comment]
I can confirm your problem. I will try to create a workaround if possible.
Thank you for your patience :)

**Comment by Krawei** [comment]
Thanks :+1:  :)

**Comment by noelkra** [comment]
I did some research and it seems that this is a VPN-specific bug. There are bugs on other VPN apps aswell that are similar to you. So I think that the only way to get rid of the bug is by temporarly deactivating Blokada for downloading the episodes at the moment.
I tried it for myself and it does not only concern Blokada but all VPN services.
I will keep trying!

Regards.
Noël

**Comment by virtadpt** [comment]
I'm seeing it, too.

Blokada: v3.4 (100500)
Blocklists: AdZHosts (yes, only one)

**Comment by hameno** [comment]
I've noticed that Pocket Casts warns about metered Wifi connection. I've contacted their support and they said, that this could be due to an active VPN. It looks like this can be fixed with P: https://developer.android.com/preview/behavior-changes#network-capabilities-vpn

**Comment by jawz101** [comment]
I've also had issues with PocketCasts and NetGuard.  @M66B says it's an issue with LineageOS but didn't leave specifics.

**Comment by ameshkov** [comment]
Might be the same as this one:
https://github.com/AdguardTeam/AdguardForAndroid/issues/1587

Quote from there:

> On Oreo, we should be calling [`VpnService.setUnderlyingNetworks`](https://developer.android.com/reference/android/net/VpnService.html#setUnderlyingNetworks(android.net.Network[])) when the tunnel is established and [`VpnService.Builder.setUnderlyingNetworks`](https://developer.android.com/reference/android/net/VpnService.Builder.html#setUnderlyingNetworks(android.net.Network[])) when we're preparing the tunnel.

Originally, it helped. Then after one of the security patches, it got back. However, it seems that there's no issue on Android P.

**Comment by jawz101** [comment]
Thanks for chiming in.  I figured bringing a few VPN app developers together might be interesting :)

**Comment by Krawei** [comment]
Any updates here?

**Comment by alexanderadam** [comment]
I had issues as well and had to add AntennaPod to the whiltelist back then.

**Comment by virtadpt** [comment]
I explicitly whitelisted Antennapod last night, and it's working as expected now.

**Comment by alexanderadam** [comment]
I'm sure it would make sense to add AntennaPod to the _default_ whitelist of blokada.

**Comment by thesuperzapper** [comment]
@kar 

Android pi, allows VPN's to pass though the fact that the underlying connection is undeterred, you have to call setUnderlyingNetworks() like is described above.

https://developer.android.com/about/versions/pie/android-9.0-changes-all#network-capabilities-vpn

This means things like podcast apps which check for an undeterred connection to download, would be able to do it with blokada running.

**Comment by Krawei** [comment]
Seems being fixed in v3.6. No need for a whitelisting anymore.

**Comment by thesuperzapper** [comment]
Are you 100% sure @Krawei, I am still getting issues.

**Comment by ZiMilio** [comment]
@Krawei @thesuperzapper This issue it's still there for me on Blokada version 3.7 (022000). It also happens on Google Podcasts App.
I've made a bit of analysis and found that Google Podcasts not downloading when Blokada is active, no matter which kind of connection I'm using (Mobile or Wifi), no matter if the app is in the list or not (following I'll post my whitelist), no matters which DNS servers I'm using (I've tried my default ISP ones, Opndns and cloudfare), no matter which blacklist is used (I've also tried to disable all blacklists). So the problem I think is related to VPN-like connection.
